kinto_http/session.py

```diff
@@ -78,16 +85,11 @@ def request(self, method, endpoint, data=None, permissions=None,
                 permissions = permissions.as_dict()
             payload['permissions'] = permissions
         if method not in ('get', 'head'):
-            payload_kwarg = 'data' if 'files' in kwargs else 'json'
-            kwargs.setdefault(payload_kwarg, payload)
-
-        # Set the default User-Agent if not already defined.
-        if 'headers' not in kwargs or kwargs['headers'] is None:
-            kwargs['headers'] = {}
-        if not isinstance(kwargs['headers'], dict):
-            raise TypeError("headers must be a dict (got {})".format(kwargs['headers']))
-
-        kwargs['headers'] = {"User-Agent": USER_AGENT, **kwargs['headers']}
+            if 'files' in kwargs:
+                kwargs.setdefault('data', payload)
+            else:
+                kwargs.setdefault('data', utils.json_dumps(payload))
```

Comment:
'json' ? (If I read the previous code `payload_kwarg = 'data' if 'files' in kwargs else 'json'`

Reply:
Yes but I changed that right?